Test with Typescript 5.9

https://devblogs.microsoft.com/typescript/announcing-typescript-5-9/ is out with some changes that could affect type checks for this library.

I originally thought that ts 5.9 is the reason why I had some type errors in https://github.com/apple/embedding-atlas/pull/15 which I eventually resolved by not allowing shared array buffers: https://github.com/apple/embedding-atlas/pull/15/commits/7d260f94f9eb07c4763a6cb12873add3881bbba0. 

Without those changes, I get the following error

```
src/lib/webgpu_renderer/renderer.ts:218:73 - error TS2322: Type 'ValueNode<Float32Array<ArrayBufferLike>>' is not assignable to type 'BufferSource | Node<BufferSource | null> | null'.
  Type 'ValueNode<Float32Array<ArrayBufferLike>>' is not assignable to type 'Node<BufferSource | null>'.
    Types of property '_value' are incompatible.
      Type 'Float32Array<ArrayBufferLike> | null' is not assignable to type 'BufferSource | null'.
        Type 'Float32Array<ArrayBufferLike>' is not assignable to type 'BufferSource | null'.
          Type 'Float32Array<ArrayBufferLike>' is not assignable to type 'ArrayBufferView<ArrayBuffer>'.
            Types of property 'buffer' are incompatible.
              Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
                Type 'SharedArrayBuffer' is missing the following properties from type 'ArrayBuffer': resizable, resize, detached, transfer, transferToFixedLength

218     [device, df.statefulDerive([device, xyDataSize, usage], gpuBuffer), inputs.xData],
                                                                            ~~~~~~~~~~~~
src/lib/webgpu_renderer/renderer.ts:222:73 - error TS2322: Type 'ValueNode<Float32Array<ArrayBufferLike>>' is not assignable to type 'BufferSource | Node<BufferSource | null> | null'.
  Type 'ValueNode<Float32Array<ArrayBufferLike>>' is not assignable to type 'Node<BufferSource | null>'.
    Types of property '_value' are incompatible.
      Type 'Float32Array<ArrayBufferLike> | null' is not assignable to type 'BufferSource | null'.
        Type 'Float32Array<ArrayBufferLike>' is not assignable to type 'BufferSource | null'.
          Type 'Float32Array<ArrayBufferLike>' is not assignable to type 'ArrayBufferView<ArrayBuffer>'.
            Types of property 'buffer' are incompatible.
              Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
                Type 'SharedArrayBuffer' is missing the following properties from type 'ArrayBuffer': resizable, resize, detached, transfer, transferToFixedLength

222     [device, df.statefulDerive([device, xyDataSize, usage], gpuBuffer), inputs.yData],
                                                                            ~~~~~~~~~~~~
src/lib/webgpu_renderer/renderer.ts:226:79 - error TS2322: Type 'ValueNode<Uint8Array<ArrayBufferLike> | null>' is not assignable to type 'BufferSource | Node<BufferSource | null> | null'.
  Type 'ValueNode<Uint8Array<ArrayBufferLike> | null>' is not assignable to type 'Node<BufferSource | null>'.
    Types of property '_value' are incompatible.
      Type 'Uint8Array<ArrayBufferLike> | null' is not assignable to type 'BufferSource | null'.
        Type 'Uint8Array<ArrayBufferLike>' is not assignable to type 'BufferSource | null'.
          Type 'Uint8Array<ArrayBufferLike>' is not assignable to type 'ArrayBufferView<ArrayBuffer>'.
            Types of property 'buffer' are incompatible.
              Type 'ArrayBufferLike' is not assignable to type 'ArrayBuffer'.
                Type 'SharedArrayBuffer' is missing the following properties from type 'ArrayBuffer': resizable, resize, detached, transfer, transferToFixedLength

226     [device, df.statefulDerive([device, categoryDataSize, usage], gpuBuffer), inputs.categoryData],
                                                                                  ~~~~~~~~~~~~~~~~~~~
```

But you are probably right that ithe ts version may not matter so feel free to close.

--- a/tests/ts5.9/package.json
+++ b/tests/ts5.9/package.json
@@ -1,0 +1,9 @@
+{
+  "scripts": {
+    "test": "tsc --version && tsc"
+  },
+  "dependencies": {
+    "@webgpu/types": "file:../..",
+    "typescript": "~5.9.0"
+  }
+}

--- a/tests/ts5.9/tsconfig.json
+++ b/tests/ts5.9/tsconfig.json
@@ -1,0 +1,4 @@
+{
+  "extends": "../tsconfig.json",
+  "include": ["../index.ts"]
+}

